x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeSourceHandler.java

```diff
@@ -215,23 +240,36 @@ void onSinkComplete() {
      * Add a remote sink as a new data source of this handler. The handler will start fetching data from this remote sink intermediately.
      *
      * @param remoteSink the remote sink
-     * @param instances  the number of concurrent ``clients`` that this handler should use to fetch pages. More clients reduce latency,
-     *                   but add overhead.
+     * @param failFast   determines how failures in this remote sink are handled:
```

Comment:
OK I think I understand this now, but I have a new question: what it means for the sink listener to handle the failure here? If I understand it correctly, if we set `failFast` to true, then the failure will close the exchange and the handler doesn't have to do anything. If we set `failFast` to false, then the exchange is not closed, but do we need e.g. to close the sink manually on the failure handler side? Or do something else? Is that going to be handled in the second patch?

Reply:
We only need to track the failure to inform the user that the query is partial and to provide details about the failure. The remote sink will be cleaned up automatically.